common/courses/states.ts

```diff
@@ -52,6 +53,27 @@ export async function subcourseOver(subcourse: Subcourse) {
     return moment(lastLecture.start).add(lastLecture.duration, 'minutes').isBefore(now);
 }
 
+/* ------------------ Course Review ----------------- */
+export async function allowCourse(course: Course, screeningComment: string | null) {
+    await prisma.course.update({ data: { screeningComment, courseState: CourseState.ALLOWED }, where: { id: course.id } });
+    logger.info(`Admin allowed (approved) Course(${course.id}) with screening comment: ${screeningComment}`);
+
+    // Usually when a new course is created, instructors also create a proper subcourse with it
+    // and then forget to publish it after it was approved. Thus we just publish during approval,
+    // assuming the subcourses are ready:
+    const subcourses = await prisma.subcourse.findMany({ where: { courseId: course.id } });
+    for (const subcourse of subcourses) {
+        if (await canPublish(subcourse)) {
+            await publishSubcourse(subcourse);
+        }
+    }
+}
+
+export async function denyCourse(course: Course, screeningComment: string | null) {
+    await prisma.course.update({ data: { screeningComment, courseState: CourseState.DENIED }, where: { id: course.id } });
+    logger.info(`Admin denied Course${course.id}) with screening comment: ${screeningComment}`);
```

Comment:
```suggestion
    logger.info(`Admin denied Course.`, {course: course.id, screeningComment: screeningComment});
```

Reply:
As long as I use "*Course(12)*" to search the logs in Datadog, I'd like to keep the "rich text" :) (but we can gradually adopt more context logging for sure)